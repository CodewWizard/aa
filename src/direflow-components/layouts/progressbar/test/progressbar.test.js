import React from "react";
import { create } from "react-test-renderer";
import ProgressBar from "../progressbar";

describe("ProgressBar component", () => {
  const schema = {
    key: "example",
    milestones: [
      {
        Label: "Step 1",
        Href: "#step-1",
        Disabled: false,
      },
      {
        Label: "Step 2",
        Href: "#step-2",
        Disabled: false,
      },
      {
        Label: "Step 3",
        Href: "#step-3",
        Disabled: true,
      },
    ],
    submitForm: "example-form",
  };
  const onHandleProgressClick = jest.fn();
  const currentPageCheck = jest.fn();
  const props = {
    schema,
    onHandleProgressClick,
    currentPageCheck,
    needContinueBtn: true,
  };

  it("should render ProgressBar component with props", () => {
    const component = create(<ProgressBar {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should update schema state when props are updated", () => {
    const component = create(<ProgressBar {...props} />);
    const newSchema = {
      key: "example-updated",
      milestones: [
        {
          Label: "Step 1",
          Href: "#step-1",
          Disabled: false,
        },
      ],
      submitForm: "example-form-updated",
    };
    component.update(<ProgressBar {...props} schema={newSchema} />);
    expect(component.root.findByType("h1").children).toEqual([
      "example-updated",
    ]);
  });

  it("should render a continue button if needContinueBtn prop is true", () => {
    const component = create(<ProgressBar {...props} />);
    expect(component.root.findByType("button").props.children).toEqual(
      "CONTINUE"
    );
  });
});
