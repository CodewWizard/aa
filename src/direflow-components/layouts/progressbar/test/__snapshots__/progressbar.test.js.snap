// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component should render ProgressBar component with props 1`] = `
<div
  className="mb-3 menuWrapper"
>
  <h1>
    example
  </h1>
  <ul
    className="quoteType p-0"
  >
    <li
      className="inactiveLink"
    >
      <a
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <label>
          <input
            disabled={true}
            name="radio"
            type="radio"
          />
          <span>
            Step 1
          </span>
        </label>
      </a>
    </li>
    <li
      className="inactiveLink"
    >
      <a
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <label>
          <input
            disabled={true}
            name="radio"
            type="radio"
          />
          <span>
            Step 2
          </span>
        </label>
      </a>
    </li>
    <li
      className="inactiveDisble"
    >
      <a
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <label>
          <input
            disabled={true}
            name="radio"
            type="radio"
          />
          <span>
            Step 3
          </span>
        </label>
      </a>
    </li>
  </ul>
  <button
    className="btn btnStyle btnSecon mt-5"
    form="example-form"
    id="pageSubmitButton"
    type="submit"
  >
    CONTINUE
  </button>
</div>
`;
